lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -1187,6 +1190,58 @@ void MayaUsdProxyShapeBase::_OnStageContentsChanged(const UsdNotice::StageConten
 void MayaUsdProxyShapeBase::_OnStageObjectsChanged(const UsdNotice::ObjectsChanged& notice)
 {
     ProxyAccessor::stageChanged(_usdAccessor, thisMObject(), notice);
+
+    // Recompute the extents of any UsdGeomBoundable that has authored extents
+    const auto& stage = notice.GetStage();
+    if (stage != getUsdStage()) {
+        TF_CODING_ERROR("We shouldn't be receiving notification for other stages than one "
+                        "returned by stage provider");
+        return;
+    }
+
+    UsdEditContext editContext(stage, stage->GetSessionLayer());
+
+    for (const auto& changedPath : notice.GetChangedInfoOnlyPaths()) {
+        if (!changedPath.IsPrimPropertyPath()) {
+            continue;
+        }
+
+        const TfToken& changedPropertyToken = changedPath.GetNameToken();
+        if (changedPropertyToken == UsdGeomTokens->extent) {
+            continue;
+        }
+
+        SdfPath          changedPrimPath = changedPath.GetPrimPath();
+        const UsdPrim&   changedPrim = stage->GetPrimAtPath(changedPrimPath);
+        UsdGeomBoundable boundableObj = UsdGeomBoundable(changedPrim);
+        if (!boundableObj) {
+            continue;
+        }
+
+        // If the attribute is not part of the primitive schema, it does not affect extents
+        const UsdPrimDefinition& primDefinition = changedPrim.GetPrimDefinition();
+        if (!primDefinition.GetSchemaAttributeSpec(changedPropertyToken)) {
+            continue;
+        }
+
+        // Ignore all attributes known to GPrim and its base classes as they
+        // are guaranteed not to affect extents:
+        static const std::unordered_set<TfToken, TfToken::HashFunctor> ignoredAttributes(
+            UsdGeomGprim::GetSchemaAttributeNames(true).cbegin(),
+            UsdGeomGprim::GetSchemaAttributeNames(true).cend());
+        if (ignoredAttributes.count(changedPropertyToken) > 0) {
+            continue;
+        }
+
+        UsdAttribute extentsAttr = boundableObj.GetExtentAttr();
+        if (extentsAttr && extentsAttr.HasValue()) {
+            VtVec3fArray extent(2);
+            if (UsdGeomBoundable::ComputeExtentFromPlugins(
+                    boundableObj, UsdTimeCode::Default(), &extent)) {
```

Comment:
Here is another challenging place. What if the extent is animated?

Reply:
I must say that reacting by notification is probably not appropriate. There is not enough context to know if the values is/was animated which forces writing a single non-animated extent value. This makes the extent completely wrong if another untouched parameter is kept animated (radius and length of a capsule, for example). There is also no information stored for Undo, and no mechanism to store the undo data of an animated extent (same issue appears in Ufe Undo handler for `TypedUsdAttribute::Set`).

We could build a set of affected boundables and wait until the Compute function of the shape node is called. At this time we might have a better overall view of the state. We still end up with animation issues: we are expected to compute the extent at the current frame, but might be forced to recompute for the whole clip duration since we have absolutely no idea about the final animated state of the extent, and more specifically, which attributes affect the extent.